src/frontend/packages/suse-extensions/src/custom/kubernetes/workloads/release/tabs/helm-release-helper.service.ts

```diff
@@ -227,14 +225,23 @@ export class HelmReleaseHelperService {
             }
           }
         }
+        // No newer release, so return the release itself if that is what was requested
+        if (returnLatest) {
+          const releaseChart = charts.find(c => c.relationships.latestChartVersion.data.version === release.chart.metadata.version);
+          return {
+            release,
+            upgrade: releaseChart.attributes,
```

Comment:
If the workload was not deployed via a connected helm, or that helm was removed, this results in a `Cannot read property of undefined ` error. Feels like the returnLatest logic should be moved into the `if (latest.isNewer(current)) {` check and the case where there's no current or newer chart handled here

Reply:
Fixed and updated so the upgrade button only shows if there is an available helm chart